docs/semgrep-ci/configuration-reference.md

```diff
@@ -19,26 +19,30 @@ While environment variables are the preferred way to configure Semgrep CI, pass
 SEMGREP_RULES="p/security-audit p/secrets"
 ```
 
-## Suppressing blocking findings or errors
+## Configuring blocking findings or errors
 
-Most CI providers block pull requests (PRs) or merge requests (MRs) on non-zero exit codes. Semgrep exits with a non-zero exit code (exit code 1) when it reports a blocking finding or there is an issue in running Semgrep (exit code 2 and above). For more information about specific Semgrep exit codes see [CLI reference](../../cli-reference/#exit-codes).
+Most CI providers block pull requests (PRs) or merge requests (MRs) when the pipeline ends with an exit code other than zero (other than status successful). However, Semgrep in CI is by default configured to suppress internal errors and only report blocking findings to stop your pipeline. Configure this behavior by the `--suppress-errors` or `--no-suppress-errors` flags.
 
-However, Semgrep in CI can run both in fail open and fail closed states. Configure the behavior of Semgrep in CI to suppress blocking findings or internal Semgrep errors blocking your pipeline by using the following options in your YAML configuration file: 
+In further detail, the default behavior of Semgrep in CI towards blocking findings and errors is as follows:
+- When the default behavior is configured, Semgrep encounters an error (exit code `2` and above) pipeline is **not** blocked. Semgrep in CI sends a crash report to a crash-reporting server and lets CI job continue. This report is anonymous and contains information that is usually sent to Semgrep App.
+- A pipeline is blocked when Semgrep reports a blocking finding (exit code `1`). Blocking findings can mean the following:
+    - Findings defined in [Rule Board](https://semgrep.dev/orgs/-/board) of Semgrep App. To avoid blocking findings, remove rules from the **Block** column of [Rule Board](https://semgrep.dev/orgs/-/board).
+    - If you do **not** use Semgrep App with Semgrep in CI, blocking findings encompass **all** Semgrep findings.
 
-- `semgrep ci` - Semgrep in CI **fails** on blocking findings, CI **fails** on internal errors.
-- `semgrep ci || [ $? != 1 ]` - Semgrep in CI **fails** on blocking findings, CI **passes** on internal errors.
+Configure and change the setup of blocking findings or errors in your CI pipeline by using the following options in your YAML configuration file:
+
+- `semgrep ci` - This is the default state. Semgrep in CI **fails** on blocking findings, CI **passes** on internal errors. Semgrep sends an anonymous crash report to a crash-reporting server and exits with exit code `0`. Optional: Define it explicitly by using `semgrep ci --suppress-errors` flag.
+- `semgrep ci --no-suppress-errors` - Semgrep in CI **fails** on blocking findings, CI **fails** on internal errors. If you use this flag, all exit codes, including internal errors, are surfaced to the CI provider.
 - `semgrep ci || true` - Semgrep in CI **passes** on blocking findings, CI **passes** on internal errors.
 
```

Comment:
@spencerdrak I think we need to add a table (without any additional details, we inform about all this above already.

```suggestion
See the following table that summarizes these configuration options:

| CI option                           | Description                         |
|-------------------------------------|-------------------------------------|
| `semgrep ci`                        | Default: CI **fails** on blocking findings, CI **passes** on internal errors.  |
| `semgrep ci --no-suppress-errors`   | CI **fails** on blocking findings, CI **fails** on internal errors.            |
| `semgrep ci || true`                | CI **passes** on blocking findings, CI **passes** on internal errors.          |

To configure these options, insert the code after the `run` key (for example `run: semgrep ci --suppress-errors` to explicitly state the default option). See the [Examples of blocking findings or errors configuration](#examples-of-blocking-findings-or-errors-configuration) below.
```